app/src/main/java/com/monkeyteam/chimpagne/model/feed/FeedAlgorithm.kt

```diff
@@ -0,0 +1,21 @@
+package com.monkeyteam.chimpagne.model.feed
+
+import com.monkeyteam.chimpagne.model.database.ChimpagneEvent
+import com.monkeyteam.chimpagne.model.location.Location
+
+val N_CLOSEST = 4
+
+fun getClosestNEvent(
+    li: List<ChimpagneEvent>,
+    myLocation: Location,
+): List<ChimpagneEvent> {
+  if (li.size <= N_CLOSEST) return li
+
+  val sortedEvents =
```

Comment:
You are sorting the whole event list so this doesn't improve performances. 
Just returning the event list sorted by distance is enough